src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -1140,6 +1140,7 @@ static void parseKeyStoreType(ConfigurationXMLStreamReader reader, final Version
         String type = null;
         String provider = null;
         Boolean wrap = null;
+        DeferredSupplier<Provider[]> providersSupplier = new DeferredSupplier<>(providers);
```

Comment:
I don't think we need to wrap the providers with a DeferredSupplier at this point, the only places we use a DeferredSupplier is if parsing may swap out the discovery approach but in this method there is no further provider parsing.